planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/scene.cpp

```diff
@@ -1565,16 +1554,44 @@ bool NormalLaneChange::get_lane_change_paths(LaneChangePaths & candidate_paths)
   return false;
 }
 
+bool NormalLaneChange::is_valid_prepare_segment(const PathWithLaneId & prepare_segment) const
+{
+  const auto & current_lanes = common_data_ptr_->lanes_ptr->current;
+  const auto & target_lanes = common_data_ptr_->lanes_ptr->target;
+
+  if (prepare_segment.points.empty()) {
+    throw std::logic_error("prepare segment is empty!");
+  }
+
+  const auto & lc_start_pose = prepare_segment.points.back().point.pose;
+
+  // TODO(Quda, Azu): Is it possible to remove these checks if we ensure prepare segment length is
+  // larger than distance to target lane start
+  if (!is_valid_start_point(common_data_ptr_, lc_start_pose)) {
+    throw std::logic_error(
+      "lane changing start point is not within the preferred lanes or its neighbors");
+  }
+
+  // lane changing start is at the end of prepare segment
+  const auto target_length_from_lane_change_start_pose =
+    utils::getArcLengthToTargetLanelet(current_lanes, target_lanes.front(), lc_start_pose);
+
+  // Check if the lane changing start point is not on the lanes next to target lanes,
+  return target_length_from_lane_change_start_pose < std::numeric_limits<double>::epsilon();
+}
+
 LaneChangePath NormalLaneChange::get_candidate_path(
   const LaneChangePhaseMetrics & prep_metrics, const LaneChangePhaseMetrics & lc_metrics,
   const PathWithLaneId & prep_segment, const std::vector<std::vector<int64_t>> & sorted_lane_ids,
-  const Pose & lc_start_pose, const double target_lane_length, const double shift_length,
-  const double next_lc_buffer, const bool is_goal_in_route) const
+  const Pose & lc_start_pose, const double shift_length) const
 {
   const auto & route_handler = *getRouteHandler();
   const auto & current_lanes = common_data_ptr_->lanes_ptr->current;
   const auto & target_lanes = common_data_ptr_->lanes_ptr->target;
+  const auto & target_lane_length = common_data_ptr_->transient_data.target_lane_length;
   const auto & forward_path_length = planner_data_->parameters.forward_path_length;
+  const auto & is_goal_in_route = common_data_ptr_->lanes_ptr->target_lane_in_goal_section;
+  const auto & next_lc_buffer = common_data_ptr_->transient_data.next_dist_buffer.min;
```

Comment:
```suggestion
  const auto is_goal_in_route = common_data_ptr_->lanes_ptr->target_lane_in_goal_section;
  const auto next_lc_buffer = common_data_ptr_->transient_data.next_dist_buffer.min;
```
trivial type, so reference is not needed.